image-registry/remove-registry-baremetal.md

```diff
@@ -0,0 +1,137 @@
+---
+title: remove-registry-baremetal
+authors:
+  - "@adambkaplan"
+reviewers:
+  - "@dmage"
+  - "@bparees"
+  - "@sdodson"
+  - "@rgolangh" # RHV
+  - "@russellb" # k8s-native infra
+approvers:
+  - "@bparees"
+  - "@smarterclayton"
+  - "@derekwaynecarr"
+creation-date: 2019-09-25
+last-updated: 2019-09-25
+status: provisional
+see-also: 
+replaces:
+superseded-by:
+---
+
+# Remove Registry on Baremetal
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+On `baremetal` platform clusters, the OpenShift image registry operator will bootstrap itself as `Removed`.
+This will allow `openshift-installer` to complete installations on this platform type.
+A separate process will be responsible for configuring storage and updating the image registry operator's state to `Managed`.
+
+## Motivation
+
+IPI installation options for Kubernetes-native infrastructure (formerly RHHI.next) and Red Hat Virtualization (RHV) have been declared a key initiative for OpenShift 4.3.
+These will configure themselves as the `baremetal` platform type.
```

Comment:
RHV has 'ovirt' platform type

Reply:
Let me try to clarify this because it is getting confusing to me. This is the list of supported platforms according to source [1]:

```
const (                                                                                                                                                                                                                                                                          
        AWSPlatformType PlatformType = "AWS"                                                                                                                                                                                                                                     
        AzurePlatformType PlatformType = "Azure"                                                                                                                                                                                                                                 
        BareMetalPlatformType PlatformType = "BareMetal"                                                                                                                                                                                                                         
        GCPPlatformType PlatformType = "GCP"                                                                                                                                                                                                                                     
        LibvirtPlatformType PlatformType = "Libvirt"                                                                                                                                                                                                                             
        OpenStackPlatformType PlatformType = "OpenStack"                                                                                                                                                                                                                         
        NonePlatformType PlatformType = "None"                                                                                                                                                                                                                                   
        VSpherePlatformType PlatformType = "VSphere"                                                                                                                                                                                                                             
        OvirtPlatformType PlatformType = "oVirt"                                                                                                                                                                                                                                 
)
```

As far as I understood everything we are discussing here refers to `BareMetalPlatformType` platform, all the others remain the same, no functional change. My guess is that you are referring to `OvirtPlatformType`, right? On this case I don't think we even try to configure the storage for [2], image registry starts as Degraded as far as I could see.

/cc @adambkaplan @dmage 

[1] https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L71
[2] https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/storage/storage.go#L87

Reply:
@rgolangh that is new information for us. I will update this enhancement so that the operator starts `Available=true, Reason=Removed` rather than `Available=false, Reason=StorageNotConfigured`